eos-tech-support/eos-reset-password

```diff
@@ -31,14 +31,49 @@ else
 fi
 
 MOUNT=`mktemp -d`
+LOOPBACK=
 
 cleanup() {
     set +e
-    umount $MOUNT
-    rmdir $MOUNT
+
+    umount "$MOUNT"
+
+    if [ -n "$LOOPBACK" ]; then
+        losetup -d "$LOOPBACK"
+        udevadm settle
+
+        umount "$MOUNT"
+    fi
+
+    rmdir "$MOUNT"
 }
 trap cleanup EXIT
 
 mount $DEVICE $MOUNT
+
+# Check if it's a Windows partition with a dual-boot Endless OS image
+LABEL=$(lsblk -o LABEL --raw --nodeps --noheadings "$DEVICE")
+ENDLESS_IMG="$MOUNT/endless/endless.img"
+if [ "$LABEL" != "ostree" ] && [ -e "$ENDLESS_IMG" ]; then
```

Comment:
`$LABEL` can legitimately be the empty string. Here's the `lsblk` output from a modern.ie Windows 7 VM:

![screenshot from 2016-11-30 11-25-21](https://cloud.githubusercontent.com/assets/86760/20750614/bc5d756e-b6ef-11e6-9d3f-f419e123bca1.png)

Notice that `sda2`, the partition to which Endless OS is installed, has an empty label.

This script could explicitly handle the case where `$DEVICE` is neither an `ostree` partition, nor contains `/endless/endless.img`, but I assume that the call to `ostree` will give a meaningful error message in that case. What do you think?